src/lib/tpm.c

```diff
@@ -22,13 +24,24 @@
 #include <tss2/tss2_esys.h>
 #include <tss2/tss2_mu.h>
 
+#include "checks.h"
 #include "openssl_compat.h"
 #include "pkcs11.h"
 #include "log.h"
 #include "mutex.h"
 #include "tcti_ldr.h"
 #include "tpm.h"
 
+/**
+ * Maps 4-byte manufacturer identifier to manufacturer name.
+ */
+static const char *TPM2_MANUFACTURER_MAP[][2] = {
```

Comment:
I thought also about this while writing.

Since PKCS11 is a 'high-level' interface, I guess users would expect something high level as vendor strings, not cryptic 4 bytes abreviations. (who is the TPM Vendor 'WEC' for example, INTC is also not too obvious.)
Since Infineon is also publicly known as IFX, I don't really have a problem in keeping the reported short handle (of IFX in my case).

We could incoporate the list of all (18) TPM vendors from the TCG Vendor Registry, https://trustedcomputinggroup.org/wp-content/uploads/Vendor_ID_Registry_0-8_clean.pdf 
which has been last updated 2015, so it does not seem to be that volatile and thus not too hard to maintain.

Since some of them are not really manufacturing any TPM2.0 chips, I would rather have the list grow, when they are really working and tested with our provider.

Maybe see being in the list as a benefit for active contributors.
I can also split out this section into a seperate patch where we can extend the discussion, as I really want to see the general thing merged.

Reply:
Im ok with the map, we can try it. Just make sure we have a default that takes the raw values and shoves them in there.